### ISSUE TITLE
Service churn prediction - train_test_val_split

### ISSUE THREAD

**Issue by DanielDaCosta** [acoullandreau/capstone_sparkify#1]
First of all, very nice medium post and github repo!

While I was reading your post on https://towardsdatascience.com/sparkify-churn-prediction-or-the-power-of-time-series-data-7b2e8f49a30e, I observed that when building your predictor for the _Service churn prediction_ you've split the train and test set using a _RandomSplit_, but as your model follows a Time Series you can't just randomly shuffle your dataset,
or can you?

As far as I know, this creates a fallacy in the results. you should not shuffle at all when working with time series.

Code line:
```
def train_test_val_split(df):
    """
        Split the dataframe into training, validation and testing set.
    """
    train, rest = df.randomSplit([0.6, 0.4], seed=42)
    validation, test = rest.randomSplit([0.5, 0.5], seed=42)
    return train, validation, test
In [126]:
train_80_cancel, validation_80_cancel, test_80_cancel = train_test_val_split(df_80_cancel_scaled)
train_70_cancel, validation_70_cancel, test_70_cancel = train_test_val_split(df_70_cancel_scaled)
train_60_cancel, validation_60_cancel, test_60_cancel = train_test_val_split(df_60_cancel_scaled)
```

Thanks in advance!

**Comment by acoullandreau** [comment]
Hello Daniel!

I agree with you, dealing with time series is not like dealing with any type of data!

However, I did not apply any model directly on the raw time series data, but on a different data structure that was built using the time series. This structure is basically a summary containing a set of features for each user each week, including the calculation of a deviation between one week and the next. And although each row contains information that is time-related (through the value of the "deviations"), there is no _order_ like in the time series, and the notion of time dependency is actually somehow blurred by the way the deviation is computed (giving a bigger weight to recent events than older events). 

It is this structure that is split into training, testing and validation sets - and as there is no order (time-wise) of the rows, they can be split randomly.
Did you, by any chance, try to split the data differently (not randomly)? I am curious to know the results you get with it :) 

Something interesting that we could look at is which feature(s) is(are) the most determinant one(s) when it comes to making the prediction, maybe none of the deviations is actually a good parameter....

Let me know if this helps!
Have a nice day,

Alexina.

**Comment by DanielDaCosta** [comment]
Hello!

Thanks for the explanation! I didn't implement your code, but I was doing something similar on a different project!

It did help!

Best Regards,

Daniel

**Comment by acoullandreau** [comment]
Great! :)
